python/pyarrow/tests/test_scalars.py

```diff
@@ -299,6 +299,46 @@ def test_timestamp_no_overflow(self):
         result = arr.to_pylist()
         assert result == timestamp_rows
 
+    def test_duration(self):
+        arr = np.array([0, 3600000000000], dtype='timedelta64[ns]')
+
+        units = ['us', 'ms', 's']
```

Comment:
Why not test nanoseconds as well?

Reply:
Nanoseconds are tested below separately, as they have a different behaviour for with or without pandas